src/interfaces/IGovernanceTimelock.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity >=0.8.9;
+
+/// @dev interface for governance timelock before execution events on dst chain
+interface IGovernanceTimelock {
+    /*/////////////////////////////////////////////////////////////////
+                                STRUCTS
+    ////////////////////////////////////////////////////////////////*/
+    struct ScheduledTransaction {
+        address target;
+        uint256 value;
+        bytes data;
+        uint256 eta;
+        bool isExecuted;
+    }
+
+    /*/////////////////////////////////////////////////////////////////
+                                EVENTS
+    ////////////////////////////////////////////////////////////////*/
+    event TransactionScheduled(uint256 indexed txId, address target, uint256 value, bytes data, uint256 eta);
+    event TransactionExecuted(uint256 indexed txId, address target, uint256 value, bytes data, uint256 eta);
```

Comment:
ScheduledTransaction is stored on the contract indexed by txId. The two events above do not necessarily need the fields except txId, which could save a bit of gas especially when the calldata is large.

Another way to save much more gas is to keep the above events and only store the hash of ScheduledTransaction on the contract. The executor needs to provide the <target, value, data, and eta> when calling then executeTransaction function.

Reply:
Agreed on the first point.

> Another way to save much more gas is to keep the above events and only store the hash of ScheduledTransaction
Yeah, that was the original spec I had. 
@sujithsomraaj and I discuss this a bit [here](https://github.com/MultiMessageAggregation/multibridge/pull/48#discussion_r1306862128). 
I briefly mentioned in that thread the reason I think it might make sense to leave it as is.